src/components/include/utils/logger.h

```diff
@@ -98,55 +98,86 @@ class Logger {
   static void SetLogger(Pimpl& impl);
   static Pimpl& GetLogger();
 
+  static void SetLoggerPtr(LoggerType& logger) {
+    logger_ = logger;
+  }
+
+  static LoggerType& GetLoggerPtr() {
+    return logger_;
+  }
+
  private:
   Logger();
   Logger(const Logger&);
   Logger& operator=(const Logger&);
   ~Logger();
 
   static Pimpl impl_;
+  static LoggerType logger_;
+};
+
+class Log {
+ public:
+  Log(const bool logs_enabled) {
+#if defined(LOG4CXX_LOGGER)
+
+    logger::Logger::InitLogger(logs_enabled, "log4cxx.properties");
+#else
+    logger::Logger::InitLogger(
+        logs_enabled, logger::LogLevel::LL_TRACE, "SmartDeviceLink.log");
```

Comment:
@malirod, agree that it is out og this PR scope.
I'm just insist to have a ticket for this functionality fixing.